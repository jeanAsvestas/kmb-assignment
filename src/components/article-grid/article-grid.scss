.article-grid {
  margin-top: 70px;
}

.article-filters {
  display: flex;
  justify-content: flex-end;
  width: 100%;
}

.articles-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
