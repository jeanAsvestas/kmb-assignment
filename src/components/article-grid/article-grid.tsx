import './article-grid.scss';
import { Article } from '../article/article';
import { SearchBox } from '../../components/searchbox/search-box';
import { SortingMenu } from '../../components/sorting-menu/sorting-menu';
import { useEffect, useState } from 'react';
import type { ArticleInterface, SortingOption } from '../../model/main.types';
import { fetchArticles } from '../../services/articles';
import Typography from '@mui/material/Typography';
import Pagination from '@mui/material/Pagination';
import Stack from '@mui/material/Stack';

export const ArticleGrid = () => {
  const [articles, setArticles] = useState<ArticleInterface[] | null>(null);
  const [page, setPage] = useState(1);
  const [sortingOption, setSortingOption] =
    useState<SortingOption>('newest-first');
  const [searchInput, setSearchInput] = useState('');

  const handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
    setPage(value);
  };

  useEffect(() => {
    setArticles(fetchArticles(page, sortingOption, searchInput));
  }, [page, sortingOption, searchInput]);

  return (
    <>
      <div className="article-grid">
        <div className="article-filters">
          <SearchBox setSearchInput={setSearchInput} />
          <SortingMenu setSortingOption={setSortingOption} />
        </div>
        <div className="articles-container">
          {articles &&
            articles.map((article: ArticleInterface, i) => {
              return <Article article={article} key={i} />;
            })}
        </div>
        <Stack spacing={2}>
          <Typography>Page: {page}</Typography>
          <Pagination count={10} page={page} onChange={handleChange} />
        </Stack>
      </div>
    </>
  );
};
