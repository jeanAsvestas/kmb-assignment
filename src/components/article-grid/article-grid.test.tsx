import TestRenderer from 'react-test-renderer';
import { ArticleGrid } from './article-grid';

it('renders the component as expected', () => {
  const component = TestRenderer.create(<ArticleGrid />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
