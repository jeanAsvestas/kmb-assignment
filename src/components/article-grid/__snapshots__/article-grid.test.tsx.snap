// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component as expected 1`] = `
<div
  className="article-grid"
>
  <div
    className="article-filters"
  >
    <form
      autoComplete="off"
      className="MuiBox-root css-xn6ruc"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="outlined-basic"
          id="outlined-basic-label"
        >
          Outlined
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="outlined-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Outlined
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <div
      className="MuiBox-root css-ece9u5"
    >
      <div
        className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          id="demo-simple-select-label"
        >
          Choose Category
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-simple-select"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Newest first
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="newest-first"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1in441m"
            >
              <span>
                Category
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="articles-container"
  />
  <div
    className="css-1p5q5e5-MuiStack-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Page: 
      1
    </p>
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            4
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            5
          </button>
        </li>
        <li>
          <div
            className="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
          >
            …
          </div>
        </li>
        <li>
          <button
            aria-label="Go to page 10"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            10
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
